lib/hal/debug.c

```diff
@@ -27,9 +37,23 @@ static const unsigned char systemFont[] =
 #include "font_unscii_16.h"
 };
 
+static void synchronizeVideoMode()
+{
+	VIDEO_MODE vm = XVideoGetMode();
+	SCREEN_WIDTH = vm.width;
+	SCREEN_HEIGHT = vm.height;
+	SCREEN_BPP = vm.bpp;
+	SCREEN_FB = XVideoGetFB();
+}
+
+static void writebackBuffers()
+{
+	asm __volatile__("sfence");
```

Comment:
I don't fully remember when it was a problem, but you can actually manage to break the displayed image if you don't do this.

Basically if you get unlucky, your pixels will not be flushed to memory and instead they remain in the CPU write-combine-buffer, where the GPU won't see them. I believe flushing them to CPU cache was enough so the GPU can see that it needs to update?

So this might be incorrect or I might have gotten lucky with `sfence` (it might have to do some other instruction). But I'm convinced that *something like this* is necessary, depending on the cache mode of the framebuffer.